zebrad/src/components/mempool.rs

```diff
@@ -85,6 +86,11 @@ impl Service<Request> for Mempool {
                 let rsp = Ok(self.storage.clone().transactions(ids)).map(Response::Transactions);
                 async move { rsp }.boxed()
             }
+            Request::RejectedTransactionsById(ids) => {
+                let rsp = Ok(self.storage.clone().rejected_transactions(ids))
+                    .map(Response::TransactionIds);
```

Comment:
Minor - confusing name: It might be clearer to name this response `RejectedTransactionIds`.

Reply:
Thanks! I named that before realizing it didn't actually return transactions. Fixed in 380c381

Reply:
Thanks!

Minor: I think it might be easier to maintain and review the code if we give the rejected transaction IDs response a different name.
```suggestion
                    .map(Response::RejectedTransactionIds);
```

Reply:
Wow, sorry, I've just realized that I misread your comment and thought you were talking about the request name.

I agree that's the best name for the response. Changed in 919948f5

For the request I'm not sure. In that commit I changed it to the same name `RejectedTransactionIds`. But I'm not sure if that's clear. Maybe `FilterRejectedTransactionIds`? We could also change to handle a single transaction (not sure if we'll ever need to handle more than one), then it could change to `IsRejected`. @dconnolly what do you think?

Reply:
Whatever we do it is important to decide now and block the functionality with the tests. It is more flexible to leave it as we have now but also more code that you need to write when using the service for simple queries. An `Request::IsRejected(id)` calling an `Storage::is_rejected() -> bool` will be simpler in that case.

We should find a use case for supporting the collection version. @dconnolly can you jump into this one ?

Reply:
Ah, this point will become moot with https://github.com/ZcashFoundation/zebra/pull/2741 since there we can directly check the mempool without a Request/Response

Reply:
Aha, I'm fine merging for now if this gets updated in that change

Reply:
Thank you both.